osgp/protocol-adapter-iec60870/osgp-protocol-adapter-iec60870/src/main/java/org/opensmartgridplatform/adapter/protocol/iec60870/application/services/LightMeasurementDeviceResponseService.java

```diff
@@ -139,17 +157,27 @@ private void sendLightSensorStatusResponse(final MeasurementGroupDto measurement
             final ResponseMetadata responseMetadata, final String errorPrefix) {
 
         this.findLightSensorStatusAndThen(measurementGroup,
-                lightSensorStatus -> this.sendLightSensorStatus(lightSensorStatus, device, responseMetadata),
+                lightSensorStatus -> this.sendOrLogLightSensorStatus(lightSensorStatus, device, responseMetadata),
                 () -> LOGGER.error("{} Light sensor status information not found for device {}", errorPrefix,
                         device.getDeviceIdentification()));
     }
 
-    private void sendLightSensorStatus(final LightSensorStatusDto lightSensorStatus, final Iec60870Device device,
+    private void sendOrLogLightSensorStatus(final LightSensorStatusDto lightSensorStatus, final Iec60870Device device,
             final ResponseMetadata responseMetadata) {
 
-        final ResponseMetadata rm = new ResponseMetadata.Builder()
-                .withCorrelationUid(responseMetadata.getCorrelationUid())
-                .withDeviceIdentification(device.getDeviceIdentification())
+        final String deviceIdentification = device.getDeviceIdentification();
+
+        Optionals.ifPresentOrElse(this.correlationUidQueuePerDevice.dequeu(deviceIdentification),
+                uid -> this.sendLightSensorStatus(uid, lightSensorStatus, deviceIdentification, responseMetadata),
+                () -> LOGGER.info("No correlation UID found for device identification {}", deviceIdentification));
```

Comment:
I am unsure how to regard the case of the missing correlation UID. I would have guessed this would warrant warning, possibly error level logging. If you made a conscious decision info level is suitable here, you can leave it as is.

Further in the review I notices correlationUids being kept with the identification of the light measurement device instead of the gateway device (alternatives that were discussed earlier). Am I correct to guess that in the current set up this logging would be triggered for all but one of the light measurement devices sharing the gateway?
If so that looks like it makes things work for now, but may be good to revisit at a later time as it seems not entirely desirable.